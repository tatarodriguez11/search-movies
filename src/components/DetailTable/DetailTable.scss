.table__container {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: var(--form-text);

  .table__poster {
    max-width: 400px;
    max-height: 600px;
    border-radius: 8px;
  }

  .table__content-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 50%;

    .table__content-container {
      display: flex;
      flex-direction: column;
      justify-content: center;

      strong {
        color: var(--form-text);
      }
    }
  }
}
