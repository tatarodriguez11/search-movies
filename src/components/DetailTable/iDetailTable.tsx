export interface DetailTableProps {
  year: string, 
  actors: string, 
  country: string, 
  language: string, 
  genre: string, 
  plot: string, 
  director: string, 
  poster: string, 
  title: string
}