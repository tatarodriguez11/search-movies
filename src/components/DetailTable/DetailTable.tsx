import { DetailTableProps } from "./iDetailTable";
import './DetailTable.scss'

export function DetailTable({year, actors, country, language, genre, plot, director, poster, title}: DetailTableProps){
  return(
    <div className="table__container">
      <img className="table__poster" src={poster} alt={title} />
      <p className="table__description">{plot}</p>
      <div className="table__content-wrapper">
        <table>
          <tr className="table__content-container">
            <td> <strong>Director:</strong>  {director}</td>
            <td><strong>Actors:</strong>  {actors}</td>
            <td><strong>Genre:</strong> {genre}</td>
            <td><strong>Country:</strong> {country}</td>
            <td><strong>Language:</strong>  {language}</td>
            <td><strong>Year:</strong> {year}</td>
          </tr>
        </table>
      </div>
    </div>
  )
}