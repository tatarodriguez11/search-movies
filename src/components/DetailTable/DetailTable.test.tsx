import { render } from "@testing-library/react";
import { DetailTable } from "./DetailTable";

describe("DetailTable", () => {
  const props = {
    year: "2022",
    actors: "Actor 1, Actor 2",
    country: "USA",
    language: "English",
    genre: "Action",
    plot: "A plot summary",
    director: "John Doe",
    poster: "https://example.com/poster.jpg",
    title: "A Movie",
  };

  it("matches the snapshot", () => {
    const { container } = render(<DetailTable {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the movie poster", () => {
    const { getByAltText } = render(<DetailTable {...props} />);
    expect(getByAltText(props.title)).toBeInTheDocument();
  });

  it("renders the movie year", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.year)).toBeInTheDocument();
  });

  it("renders the movie actors", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.actors)).toBeInTheDocument();
  });

  it("renders the movie genre", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.genre)).toBeInTheDocument();
  });

  it("renders the movie country", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.country)).toBeInTheDocument();
  });

  it("renders the movie language", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.language)).toBeInTheDocument();
  });

  it("renders the movie plot", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.plot)).toBeInTheDocument();
  });

  it("renders the movie director", () => {
    const { getByText } = render(<DetailTable {...props} />);
    expect(getByText(props.director)).toBeInTheDocument();
  });
});