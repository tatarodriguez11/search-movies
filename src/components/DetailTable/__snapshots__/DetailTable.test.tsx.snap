// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailTable matches the snapshot 1`] = `
<div>
  <div
    class="table__container"
  >
    <img
      alt="A Movie"
      class="table__poster"
      src="https://example.com/poster.jpg"
    />
    <p
      class="table__description"
    >
      A plot summary
    </p>
    <div
      class="table__content-wrapper"
    >
      <table>
        <tr
          class="table__content-container"
        >
          <td>
             
            <strong>
              Director:
            </strong>
              
            John Doe
          </td>
          <td>
            <strong>
              Actors:
            </strong>
              
            Actor 1, Actor 2
          </td>
          <td>
            <strong>
              Genre:
            </strong>
             
            Action
          </td>
          <td>
            <strong>
              Country:
            </strong>
             
            USA
          </td>
          <td>
            <strong>
              Language:
            </strong>
              
            English
          </td>
          <td>
            <strong>
              Year:
            </strong>
             
            2022
          </td>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
